Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyPostedInvoiceExport.Codeunit.al

```diff
@@ -0,0 +1,431 @@
+namespace Microsoft.Integration.Shopify;
+
+using Microsoft.Sales.History;
+using Microsoft.Finance.GeneralLedger.Setup;
+using Microsoft.Utilities;
+using Microsoft.Sales.Receivables;
+
+/// <summary>
+/// Codeunit Shpfy Posted Invoice Export" (ID 30316).
+/// </summary>
+codeunit 30316 "Shpfy Posted Invoice Export"
+{
+    Access = Internal;
+    TableNo = "Sales Invoice Header";
+
+    var
+        ShpfyShop: Record "Shpfy Shop";
+        ShpfyDraftOrdersAPI: Codeunit "Shpfy Draft Orders API";
+        ShpfyFulfillmentAPI: Codeunit "Shpfy Fulfillment API";
+        ShpfyJsonHelper: Codeunit "Shpfy Json Helper";
+
+    trigger OnRun()
+    begin
+        ExportPostedSalesInvoiceToShopify(Rec);
+    end;
+
+    /// <summary> 
+    /// Sets a global shopify shop to be used for posted invoice export.
+    /// </summary>
+    /// <param name="NewShopCode">Shopify shop code to be set.</param>
+    internal procedure SetShop(NewShopCode: Code[20])
+    begin
+        ShpfyShop.Get(NewShopCode);
+        ShpfyDraftOrdersAPI.SetShop(ShpfyShop.Code);
+        ShpfyFulfillmentAPI.SetShop(ShpfyShop.Code);
+    end;
+
+    /// <summary>
+    /// Exports provided posted sales invoice to shopify.
+    /// </summary>
+    /// <remarks>
+    /// If the posted sales invoice isn't exportable, the shopify order id is set to -2. 
+    /// If shopify order creation fails, the id is set to -1.
+    /// </remarks>
+    /// <param name="SalesInvoiceHeader">Posted sales invoice to be exported.</param>
+    internal procedure ExportPostedSalesInvoiceToShopify(SalesInvoiceHeader: Record "Sales Invoice Header")
+    var
+        TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        TempShpfyOrderLine: Record "Shpfy Order Line" temporary;
+        ShpfyFulfillmentAPI: Codeunit "Shpfy Fulfillment API";
+        DraftOrderId: BigInteger;
+        ShpfyOrderTaxLines: Dictionary of [Text, Decimal];
+        JFulfillments: JsonToken;
+        JResponse: JsonToken;
+    begin
+        if not IsInvoiceExportable(SalesInvoiceHeader) then begin
+            SetSalesInvoiceShopifyOrderInformation(SalesInvoiceHeader, -2, '');
+            exit;
+        end;
+
+        MapPostedSalesInvoiceData(SalesInvoiceHeader, TempShpfyOrderHeader, TempShpfyOrderLine, ShpfyOrderTaxLines);
+
+        DraftOrderId := ShpfyDraftOrdersAPI.CreateDraftOrder(TempShpfyOrderHeader, TempShpfyOrderLine, ShpfyOrderTaxLines);
+        JResponse := ShpfyDraftOrdersAPI.CompleteDraftOrder(DraftOrderId);
+        JFulfillments := ShpfyFulfillmentAPI.GetFulfillmentOrders(
```

Comment:
Shouldn't this be inside the 
if IsSuccess then?

Reply:
I think you're right, that would make more sense